order-book/src/actors/maker.rs

```diff
@@ -0,0 +1,88 @@
+use crate::storage::{
+    common_storage::{Percent, MAX_PERCENT},
+    order::{
+        Order, OrderDuration, OrderId, OrderStatus, DAY_IN_SECONDS, HOUR_IN_SECONDS,
+        MINUTE_IN_SECONDS,
+    },
+};
+
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait MakerModule:
+    crate::storage::order::OrderModule + crate::events::EventsModule + crate::pause::PauseModule
+{
+    #[payable("*")]
+    #[endpoint(createOrder)]
+    fn create_order(
+        &self,
+        output_token: TokenIdentifier,
+        min_total_output: BigUint,
+        order_duration: OrderDuration,
+        opt_executor_fee: OptionalValue<Percent>,
+    ) -> OrderId {
+        self.require_not_paused();
+
+        require!(
+            output_token.is_valid_esdt_identifier(),
+            "Invalid ESDT specified for output token"
+        );
+
+        let executor_fee = match opt_executor_fee {
+            OptionalValue::Some(fee) => fee,
+            OptionalValue::None => 0,
+        };
+        require!(executor_fee <= MAX_PERCENT, "Invalid executor fee");
+
+        let current_timestamp = self.blockchain().get_block_timestamp();
+        let expiration_timestamp = match &order_duration {
+            OrderDuration::Minutes(minutes) => *minutes as u64 * MINUTE_IN_SECONDS,
+            OrderDuration::Hours(hours) => *hours as u64 * HOUR_IN_SECONDS,
+            OrderDuration::Days(days) => *days as u64 * DAY_IN_SECONDS,
+        };
+        require!(
```

Comment:
This doesn't really make sense. We should compute the expiration timestamp as current_timestamp + order duration.